src/Features/Core/Portable/CodeStyle/AbstractCodeStyleProvider.Fixing.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Editing;
+
+namespace Microsoft.CodeAnalysis.CodeStyle
+{
+    // This part contains all the logic for hooking up the CodeFixProvider to the CodeStyleProvider.
+    // All the code in this part is an implementation detail and is intentionally private so that
+    // subclasses cannot change anything.  All code relevant to subclasses relating to fixing is
+    // contained in AbstractCodeStyleProvider.cs
+
+    internal abstract partial class AbstractCodeStyleProvider<TOptionKind, TCodeStyleProvider>
+    {
+        private async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var document = context.Document;
+            var diagnostic = context.Diagnostics[0];
+            var cancellationToken = context.CancellationToken;
+
+            var codeFixes = await ComputeCodeActionsAsync(
+                document, diagnostic, cancellationToken).ConfigureAwait(false);
+            context.RegisterFixes(codeFixes, context.Diagnostics);
+        }
+
+        public abstract class CodeFixProvider : SyntaxEditorBasedCodeFixProvider
+        {
+            public readonly TCodeStyleProvider _codeStyleProvider;
+
+            protected CodeFixProvider()
+            {
+                _codeStyleProvider = new TCodeStyleProvider();
+                FixableDiagnosticIds = ImmutableArray.Create(_codeStyleProvider._descriptorId);
```

Comment:
How about fixers that can fix multiple diagnostic IDs?

Reply:
You're right. We'll probably need to handle that at some point.  I could do that as part of this PR, or wait until we need it.  I'd prefer hte latter :)